libparsec/crates/platform_device_loader/src/native/mod.rs

```diff
@@ -280,25 +259,11 @@ pub async fn save_device(
         }
 
         DeviceAccessStrategy::Password { key_file, password } => {
-            let salt = SecretKey::generate_salt();
-            let opslimit = ARGON2ID_DEFAULT_OPSLIMIT;
-            let memlimit_kb = ARGON2ID_DEFAULT_MEMLIMIT_KB;
-            let parallelism = ARGON2ID_DEFAULT_PARALLELISM;
-
-            let key = SecretKey::from_argon2id_password(
-                password,
-                &salt,
-                opslimit,
-                memlimit_kb,
-                parallelism,
-            )
-            .expect("Salt has the correct length");
-
-            let ciphertext = {
-                let cleartext = Zeroizing::new(device.dump());
-                let ciphertext = key.encrypt(&cleartext);
-                ciphertext.into()
-            };
+            let key_algo = super::new_default_pbkdf_algo();
+            let key = super::secret_key_from_password(password, &key_algo)
+                .expect("Salt has the correct length");
```

Comment:
Message in expect should be for the error case.
```suggestion
                .expect("Salt does not have the correct length");
```
or more generic since errors could come from things other than the salt:
```suggestion
                .expect("Failed to derive key from password");
```